src/v/cloud_storage/remote_partition.cc

```diff
@@ -1026,22 +990,19 @@ ss::future<remote_partition::finalize_result> remote_partition::finalize(
  *    S3, not the state of the archival metadata stm (which can be out
  *    of date if e.g. we were not the leader)
  */
-ss::future<> remote_partition::erase(
-  ss::abort_source& as,
-  raft::vnode self,
-  raft::group_configuration raft_config) {
-    if (!raft_config.is_voter(self)) {
-        // Learners are excluded from deletion, because they have a high
-        // risk of having some out of date state, and because there will
-        // always be some voter peers to do the work.
-        co_return;
-    }
```

Comment:
>A reviewer will likely notice what appears to be two separate
>mechanisms for determining if the current node is a voter or not. It is
>not obvious that these two mechanisms (APIs) are identical. If so, a
>future clean-up may be useful to remove the ambiguity in the the voter
>check.

It looks like one takes into account voters of the old config during joint consensus. I can't think of a reason it'd be important for erase() to consider the old config before erasing. +1 to just picking one (I'd think perhaps just stick with the voter_position one since we'll need to do the lookup there anyway?)

Reply:
I don't think I can make that decision. So I'll rely on @mmaslankaprv and @Lazin to chime in on if those checks can be changed safely (though, I agree with your assessment, and can't think of any reason).

Reply:
That change looks safe for me also I'm not sure it was a great idea to put this into the `remote_partition` which is essentially a read path of the tiered storage.